pkg/providers/ovirt/validation/validators/network-mapping-validator.go

```diff
@@ -0,0 +1,154 @@
+package validators
+
+import (
+	"fmt"
+
+	"kubevirt.io/client-go/kubecli"
+
+	netv1 "github.com/k8snetworkplumbingwg/network-attachment-definition-client/pkg/apis/k8s.cni.cncf.io/v1"
+	v2vv1alpha1 "github.com/kubevirt/vm-import-operator/pkg/apis/v2v/v1alpha1"
+	ovirtsdk "github.com/ovirt/go-ovirt"
+)
+
+//NetworkAttachmentDefinitionProvider retrieves NetworkAttachmentDefinition for given name and optional namespace
+type NetworkAttachmentDefinitionProvider interface {
+	Find(name string, namespace string) (*netv1.NetworkAttachmentDefinition, error)
+}
+
+//NetworkMappingValidator provides network mappings validation logic
+type NetworkMappingValidator struct {
+	provider NetworkAttachmentDefinitionProvider
+}
+
+//NewNetworkMappingValidator creates new NetworkMappingValidator that will use provided KubevirtClient
+func NewNetworkMappingValidator(kubevirt kubecli.KubevirtClient) NetworkMappingValidator {
+	return NetworkMappingValidator{
+		provider: &NetworkAttachmentDefinitions{
+			Client: kubevirt.NetworkClient(),
+		},
+	}
+}
+
+//ValidateNetworkMapping validates network mapping
+func (v *NetworkMappingValidator) ValidateNetworkMapping(nics []*ovirtsdk.Nic, mapping *[]v2vv1alpha1.ResourceMappingItem, defaultNamespace string) []ValidationFailure {
+	var failures []ValidationFailure
+	// Check whether mapping for network is required and was provided
+	if mapping == nil && len(nics) > 0 {
+		failures = append(failures, ValidationFailure{
+			ID:      NetworkMappingID,
+			Message: "Network mapping is missing",
+		})
+		return failures
+	}
+	// Get all networks needed by the VM as slice of sources
+	requiredNetworks := getRequiredNetworks(nics)
+	mapByID := make(map[string]v2vv1alpha1.ResourceMappingItem)
+	mapByName := make(map[string]v2vv1alpha1.ResourceMappingItem)
+	// Map source id and name to ResourceMappingItem
+	for _, item := range *mapping {
+		if item.Source.ID != nil {
+			mapByID[*item.Source.ID] = item
+		}
+		if item.Source.Name != nil {
+			mapByName[*item.Source.Name] = item
+		}
+	}
+	requiredTargetsSet := make(map[v2vv1alpha1.ObjectIdentifier]*string)
+	// Validate that all vm networks are mapped and populate requiredTargetsSet for target existence check
+	for _, network := range requiredNetworks {
+		if network.ID != nil {
+			item, found := mapByID[*network.ID]
+			if found {
+				requiredTargetsSet[item.Target] = item.Type
+				continue
+			}
+		}
+		if network.Name != nil {
+			item, found := mapByName[*network.Name]
+			if found {
+				requiredTargetsSet[item.Target] = item.Type
+				continue
+			}
+		}
+		failures = append(failures, ValidationFailure{
+			ID:      NetworkMappingID,
+			Message: fmt.Sprintf("Required source network '%s' lacks mapping", ToLoggableID(network.ID, network.Name)),
+		})
+	}
+
+	// Validate that all target networks needed by the VM exist in k8s
```

Comment:
the rich client took care to update the full vm entity with its nics, and vnic profile for each nic and the network for each vnic profile.

so the vm spec creation could simply iterate over the vm nics and find the matching target network while the source name/id are available without a need to figure them again, right ?

Reply:
Yes, that's correct.